File: PO_java/program/Organisms/Animals/Sheep.java
package projekt.program.Organisms.Animals;

import projekt.program.Organism;
import projekt.program.Organisms.Animal;
import projekt.program.World;

import java.util.Random;

public class Sheep extends Animal {

    private static final int SHEEP_INITIATIVE = 4; // Example value
    private static final int SHEEP_STRENGTH = 4; // Example value

    public Sheep(World world, int x, int y) {
        super(world, x, y, SHEEP_INITIATIVE, SHEEP_STRENGTH, 'O');
        System.out.println("Sheep created");
    }

    @Override
    public String getName() {
        return "Sheep";
    }

    @Override
    protected Organism createChild(int x, int y) {
        return new Sheep(this.world, x, y);
    }
}

File: PO_py/program/Organisms/Animals/Sheep.py
from program.Organisms.Animal import Animal


class Sheep(Animal):
    SHEEP_INITIATIVE = 4  # Example value
    SHEEP_STRENGTH = 4  # Example value

    def __init__(self, world, x, y):
        super().__init__(world, x, y, self.SHEEP_INITIATIVE, self.SHEEP_STRENGTH, 'O')
        print("Sheep created")

    def get_name(self):
        return "Sheep"

    def create_child(self, x, y):
        return Sheep(self.world, x, y)
